src/Drivers/Decorator.php

```diff
@@ -105,29 +108,65 @@ public function discover($namespace = '\\App\\Features', $path = null)
     public function define($feature, $resolver = null): void
     {
         if (func_num_args() === 1) {
-            [$feature, $resolver] = [
-                $this->container->make($feature)->name ?? $feature,
-                fn ($scope) => with($this->container[$feature], fn ($featureClass) => method_exists($featureClass, 'resolve')
-                    ? $featureClass->resolve($scope)
-                    : $featureClass($scope)),
-            ];
-        }
-
-        if (is_string($resolver) || ! is_callable($resolver)) {
-            $resolver = fn () => $resolver;
+            [$feature, $resolver] = with($this->container[$feature], fn ($featureClass) => [
+                $featureClass->name ?? $feature,
+                method_exists($featureClass, 'resolve') ? $featureClass->resolve(...) : $featureClass(...),
+            ]);
```

Comment:
This change means that the feature class stays in-memory once it is defined and isn't re-resolved each time it is checked.

Depending on the context, this may actually be a performance gain, as the class isn't being re-resolved.

It is only really a concern if the user is mutating the class each time it is invoked - but there is an event for tracking that kind of thing.